framework/src/postprocessors/PointValue.C

```diff
@@ -34,66 +36,34 @@ validParams<PointValue>()
 
 PointValue::PointValue(const InputParameters & parameters)
   : GeneralPostprocessor(parameters),
-    _var(_subproblem.getVariable(_tid, parameters.get<VariableName>("variable"))),
-    _u(_var.sln()),
-    _mesh(_subproblem.mesh()),
-    _point_vec(1, getParam<Point>("point")),
-    _value(0),
-    _root_id(0),
-    _elem_id(DofObject::invalid_id)
+    _var_number(_subproblem.getVariable(_tid, parameters.get<VariableName>("variable")).number()),
+    _system(_subproblem.getSystem(getParam<VariableName>("variable"))),
+    _point(getParam<Point>("point")),
+    _value(0)
 {
 }
 
 void
 PointValue::execute()
 {
-  // Locate the element and store the id
-  // We can't store the actual Element pointer here b/c PointLocatorBase returns a const Elem *
-  std::unique_ptr<PointLocatorBase> pl = _mesh.getPointLocator();
-  pl->enable_out_of_mesh_mode();
-  const Elem * elem = (*pl)(_point_vec[0]);
+  _value = _system.point_value(_var_number, _point, false);
 
-  // Store the element id and processor id that owns the located element
-  if (elem)
-  {
-    _elem_id = elem->id();
-    _root_id = elem->processor_id();
-  }
-  else
+  /**
+   * If we get exactly zero, we don't know if the locator couldn't find an element, or
+   * if the solution is truly zero, more checking is needed.
+   */
+  if (MooseUtils::absoluteFuzzyEqual(_value, 0.0))
```

Comment:
@roystgnr - I don't love this implementation but libMesh's `point_value` method can't tell me the difference between an actual zero and a failed locator. Instead I get an assertion or some other behavior that I can control as the caller.

Maybe we should consider throwing when we aren't in "out of mesh mode" AND the user requests a point out of range?

Reply:
We can't really change the existing implementation without potentially messing with other users...

But we're overdue for a new ```System::point_*()``` API anyways, to update them to handle vector-valued elements like we did for the ```FEMContext::point_*()``` APIs long ago.  Assuming we do things the same way, then we'll be returning data via non-const reference argument (to allow for easy template type deduction) anyway, so instead of throwing an exception we could just leave the input argument unchanged if there's a locator failure.  Then if you want to test for failure you just initialize with NaN or something and test for it after the call, and if you want an "out of mesh value" the way we have with MeshFunction then you can just initialize with that.

Reply:
OK - that sounds like a reasonable path forward.